2013-09-13 18:03:11-0700 [-] Log opened.
2013-09-13 18:03:11-0700 [-] twistd 13.1.0 (/usr/local/opt/python/bin/python2.7 2.7.5) starting up.
2013-09-13 18:03:11-0700 [-] reactor class: twisted.internet.selectreactor.SelectReactor.
2013-09-13 18:03:11-0700 [-] ChatFactory starting on 25683
2013-09-13 18:03:11-0700 [-] Starting factory <server.ChatFactory instance at 0x102167248>
2013-09-13 18:03:19-0700 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 59034))
2013-09-13 18:03:46-0700 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 59039))
2013-09-13 18:29:11-0700 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 59098))
2013-09-13 18:29:19-0700 [ChatProtocol,2,127.0.0.1] Unhandled Error
	Traceback (most recent call last):
	  File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 137, in maybeDeferred
	    result = f(*args, **kw)
	  File "/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py", line 1036, in doit
	    makeResponseFor).addErrback(
	  File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 304, in addCallback
	    callbackKeywords=kw)
	  File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 293, in addCallbacks
	    self._runCallbacks()
	--- <exception caught here> ---
	  File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 575, in _runCallbacks
	    current.result = callback(current.result, *args, **kw)
	  File "/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py", line 1034, in makeResponseFor
	    originalFailure)
	twisted.protocols.amp.BadLocalReturn: <bound method ChatProtocol.sendMessage of <ChatProtocol at 0x1021bd690>> returned None and <class 'server.SendMessage'> could not serialize it Traceback: <type 'exceptions.AttributeError'>: 'NoneType' object has no attribute 'copy'
	/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py:1036:doit
	/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py:304:addCallback
	/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py:293:addCallbacks
	/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py:575:_runCallbacks
	--- <exception caught here> ---
	/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py:1025:makeResponseFor
	/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py:1660:makeResponse
	/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py:2533:_objectsToStrings
	
	
2013-09-13 18:31:32-0700 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 59105))
2013-09-13 18:31:32-0700 [ChatProtocol,3,127.0.0.1] Unhandled Error
	Traceback (most recent call last):
	  File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 137, in maybeDeferred
	    result = f(*args, **kw)
	  File "/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py", line 1036, in doit
	    makeResponseFor).addErrback(
	  File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 304, in addCallback
	    callbackKeywords=kw)
	  File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 293, in addCallbacks
	    self._runCallbacks()
	--- <exception caught here> ---
	  File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 575, in _runCallbacks
	    current.result = callback(current.result, *args, **kw)
	  File "/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py", line 1034, in makeResponseFor
	    originalFailure)
	twisted.protocols.amp.BadLocalReturn: <bound method ChatProtocol.sendMessage of <ChatProtocol at 0x1021bd850>> returned None and <class 'server.SendMessage'> could not serialize it Traceback: <type 'exceptions.AttributeError'>: 'NoneType' object has no attribute 'copy'
	/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py:1036:doit
	/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py:304:addCallback
	/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py:293:addCallbacks
	/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py:575:_runCallbacks
	--- <exception caught here> ---
	/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py:1025:makeResponseFor
	/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py:1660:makeResponse
	/usr/local/lib/python2.7/site-packages/twisted/protocols/amp.py:2533:_objectsToStrings
	
	
2013-09-13 18:32:54-0700 [-] Received SIGTERM, shutting down.
2013-09-13 18:32:54-0700 [-] (TCP Port 25683 Closed)
2013-09-13 18:32:54-0700 [-] Stopping factory <server.ChatFactory instance at 0x102167248>
2013-09-13 18:32:54-0700 [-] Main loop terminated.
2013-09-13 18:32:54-0700 [-] Server Shut Down.
2013-09-13 18:32:59-0700 [-] Log opened.
2013-09-13 18:32:59-0700 [-] twistd 13.1.0 (/usr/local/opt/python/bin/python2.7 2.7.5) starting up.
2013-09-13 18:32:59-0700 [-] reactor class: twisted.internet.selectreactor.SelectReactor.
2013-09-13 18:32:59-0700 [-] ChatFactory starting on 25683
2013-09-13 18:32:59-0700 [-] Starting factory <server.ChatFactory instance at 0x10abce248>
2013-09-13 18:33:49-0700 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 59114))
2013-09-13 18:44:40-0700 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 59248))
2013-10-03 11:25:26-0300 [-] Log opened.
2013-10-03 11:25:26-0300 [-] twistd 13.1.0 (/usr/bin/python 2.7.2) starting up.
2013-10-03 11:25:26-0300 [-] reactor class: twisted.internet.selectreactor.SelectReactor.
2013-10-03 11:25:26-0300 [-] ChatFactory starting on 25683
2013-10-03 11:25:26-0300 [-] Starting factory <server.ChatFactory instance at 0x101ae5d88>
2013-10-03 11:25:34-0300 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 53555))
2013-10-03 17:02:13-0300 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 58475))
2013-10-03 17:04:52-0300 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 58486))
2013-10-03 17:06:47-0300 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 58496))
2013-10-03 17:15:38-0300 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 58578))
2013-10-03 17:16:03-0300 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 58584))
2013-10-03 17:18:32-0300 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 58600))
2013-10-03 17:19:06-0300 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 58606))
2013-10-03 17:21:41-0300 [server.ChatFactory] ChatProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 25683) PEER:IPv4Address(TCP, '127.0.0.1', 58617))
